livy/src/main/java/org/apache/zeppelin/livy/BaseLivyInterprereter.java

```diff
@@ -299,8 +299,11 @@ private InterpreterResult appendSessionExpire(InterpreterResult result, boolean
 
   private InterpreterResult getResultFromStatementInfo(StatementInfo stmtInfo,
                                                        boolean displayAppInfo) {
-    if (stmtInfo.output.isError()) {
+    if (stmtInfo.output != null && stmtInfo.output.isError()) {
       return new InterpreterResult(InterpreterResult.Code.ERROR, stmtInfo.output.evalue);
+    } else if (stmtInfo.isCancelled()) {
+      // corner case, output might be null if it is cancelled.
+      return new InterpreterResult(InterpreterResult.Code.ERROR, "Job is cancelled");
     } else {
       //TODO(zjffdu) support other types of data (like json, image and etc)
```

Comment:
hmm, what is stmtInfo.output  == null && stmtInfo.isCancelled() == false? should we check the else branch that output != null?